pkg/datastore/helper.go

```diff
@@ -66,33 +66,35 @@ func FromEnv(l *logrus.Entry, authorizer authorizer.Authorizer, instancer author
 		SSL_MODE_ENV_VAR,
 	} {
 		if _, isPresent := os.LookupEnv(envVar); !isPresent {
-			err = ErrMissingEnvVar.WithValue(ENV_VAR, envVar)
-			l.Error(err)
-			return nil, err
+			panic(ErrMissingEnvVar.WithValue(ENV_VAR, envVar))
 		}
 
 		if envVarValue := strings.TrimSpace(os.Getenv(envVar)); len(envVarValue) == 0 {
-			err = ErrMissingEnvVar.WithValue(ENV_VAR, envVar).WithValue(VALUE, os.Getenv(envVar))
-			l.Error(err)
-			return nil, err
+			panic(ErrMissingEnvVar.WithValue(ENV_VAR, envVar).WithValue(VALUE, os.Getenv(envVar)))
 		}
 	}
 
 	var cfg DBConfig
+	var err error
 
 	cfg.host = strings.TrimSpace(os.Getenv(DB_HOST_ENV_VAR))
 	// Ensure port number is valid
 	if cfg.port, err = strconv.Atoi(strings.TrimSpace(os.Getenv(DB_PORT_ENV_VAR))); err != nil {
-		err = ErrInvalidPortNumber.Wrap(err).WithValue(ENV_VAR, DB_PORT_ENV_VAR).WithValue(VALUE, os.Getenv(DB_PORT_ENV_VAR))
-		l.Error(err)
-		return nil, err
+		panic(ErrInvalidPortNumber.Wrap(err).WithValue(ENV_VAR, DB_PORT_ENV_VAR).WithValue(VALUE, os.Getenv(DB_PORT_ENV_VAR)))
 	}
 	cfg.username = strings.TrimSpace(os.Getenv(DB_ADMIN_USERNAME_ENV_VAR))
 	cfg.password = strings.TrimSpace(os.Getenv(DB_ADMIN_PASSWORD_ENV_VAR))
-	cfg.dbName = strings.TrimSpace(os.Getenv(DB_NAME_ENV_VAR))
+	cfg.dbName = strings.ToLower(strings.TrimSpace(os.Getenv(DB_NAME_ENV_VAR)))
 	cfg.sslMode = strings.TrimSpace(os.Getenv(SSL_MODE_ENV_VAR))
 
-	return FromConfig(l, authorizer, instancer, cfg)
+	if dbName != "" {
+		cfg.dbName = strings.ToLower(dbName)
+	}
+	return cfg
+}
+
+func FromEnv(l *logrus.Entry, authorizer authorizer.Authorizer, instancer authorizer.Instancer) (d DataStore, err error) {
+	return FromConfig(l, authorizer, instancer, ConfigFromEnv(""))
 }
 
 func FromConfig(l *logrus.Entry, authorizer authorizer.Authorizer, instancer authorizer.Instancer, cfg DBConfig) (d DataStore, err error) {
```

Comment:
`FromConfig` never returns an error, but `error` is one of its output parameters

Reply:
I will leave it for now, not a good idea to keep breaking interface signatures